js/src/otelAppInsights.js

```diff
@@ -0,0 +1,62 @@
+const { app } = require('@azure/functions');
+const { AzureMonitorLogExporter, AzureMonitorTraceExporter } = require('@azure/monitor-opentelemetry-exporter');
+const { context: otelContext, propagation } = require('@opentelemetry/api');
+const { SeverityNumber } = require('@opentelemetry/api-logs');
+const { getNodeAutoInstrumentations, getResourceDetectors } = require('@opentelemetry/auto-instrumentations-node');
+const { registerInstrumentations } = require('@opentelemetry/instrumentation');
+const { detectResourcesSync } = require('@opentelemetry/resources');
+const { LoggerProvider, SimpleLogRecordProcessor } = require('@opentelemetry/sdk-logs');
+const { NodeTracerProvider, SimpleSpanProcessor } = require('@opentelemetry/sdk-trace-node');
+
+const resource = detectResourcesSync({ detectors: getResourceDetectors() });
+
+const tracerProvider = new NodeTracerProvider({ resource });
+tracerProvider.addSpanProcessor(new SimpleSpanProcessor(new AzureMonitorTraceExporter()));
+tracerProvider.register();
+
+const loggerProvider = new LoggerProvider({ resource });
+loggerProvider.addLogRecordProcessor(new SimpleLogRecordProcessor(new AzureMonitorLogExporter()));
+
+registerInstrumentations({ tracerProvider, loggerProvider, instrumentations: [getNodeAutoInstrumentations()] });
+
+// NOTE: The below code will soon be a part of a new package `@opentelemetry/instrumentation-azure-functions`
+// See here for more info: https://github.com/Azure/azure-functions-nodejs-library/issues/245
+app.setup({ capabilities: { WorkerOpenTelemetryEnabled: true } });
+
+const logger = loggerProvider.getLogger('default');
+app.hook.log((context) => {
+    logger.emit({
+        body: context.message,
+        severityNumber: toOtelSeverityNumber(context.level),
+        severityText: context.level,
+    });
+});
+
+app.hook.preInvocation((context) => {
+    context.functionHandler = otelContext.bind(
+        propagation.extract(otelContext.active(), {
+            traceparent: context.invocationContext.traceContext.traceParent,
+            tracestate: context.invocationContext.traceContext.traceState,
+        }),
+        context.functionHandler
+    );
+});
+
+function toOtelSeverityNumber(level) {
```

Comment:
I think it would be nice to extract this function out, but it's not a big deal for these samples.

Reply:
Right, since these are samples for different scenarios we need to copy it for each scenario. And yes this will be "extracted out" once we have the community package that replaces all this boilerplate